src/Core/Form/IdentifiableObject/DataHandler/SpecificPriceFormDataHandler.php

```diff
@@ -0,0 +1,134 @@
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(
+        CommandBusInterface $commandBus
+    ) {
+        $this->commandBus = $commandBus;
+    }
+
+    public function create(array $data): int
+    {
+        $command = new AddProductSpecificPriceCommand(
+            (int) $data['product_id'],
+            $data['reduction']['type'],
+            (string) $data['reduction']['value'],
+            (bool) $data['include_tax'],
+            (string) $data['price'],
+            (int) $data['from_quantity']
+        );
+
+        $command
+            ->setCurrencyId((int) $data['currency_id'])
+            ->setCountryId((int) $data['country_id'])
+            ->setGroupId((int) $data['group_id'])
+            ->setCustomerId($this->getCustomerId($data))
+        ;
+        if (isset($data['shop_id'])) {
+            $command->setShopId((int) $data['shop_id']);
+        }
+
+        /** @var SpecificPriceId $specificPriceId */
+        $specificPriceId = $this->commandBus->handle($command);
+
+        return $specificPriceId->getValue();
+    }
+
+    public function update($id, array $data): void
+    {
+        $command = new EditProductSpecificPriceCommand((int) $id);
+        $command->setCustomerId($this->getCustomerId($data));
+
+        if (null !== $data['currency_id']) {
+            $command->setCurrencyId((int) $data['currency_id']);
+        }
+        if (null !== $data['country_id']) {
+            $command->setCountryId((int) $data['country_id']);
+        }
+        if (null !== $data['group_id']) {
+            $command->setGroupId((int) $data['group_id']);
+        }
+        if (null !== $data['from_quantity']) {
+            $command->setFromQuantity((int) $data['from_quantity']);
+        }
+
+        if (null !== $data['leave_initial_price']) {
+            $command->setLeaveInitialPrice((bool) $data['leave_initial_price']);
+        }
+
+        if (null !== $data['price']) {
+            $command->setPrice((string) $data['price']);
+        }
+
+        if (isset($data['date_range']['from'])) {
+            $command->setDateTimeFrom(DateTime::buildNullableDateTime($data['date_range']['from']));
+        }
+        if (isset($data['date_range']['to'])) {
+            $command->setDateTimeTo(DateTime::buildNullableDateTime($data['date_range']['to']));
+        }
+        if (null !== $data['reduction']) {
+            $command->setReduction((string) $data['reduction']['type'], (string) $data['reduction']['value']);
+        }
+        if (null !== $data['include_tax']) {
+            $command->setIncludesTax((bool) $data['include_tax']);
+        }
+
+        $this->commandBus->handle($command);
+    }
+
+    /**
+     * @param array<string, mixed> $data
+     *
+     * @return int
+     */
+    private function getCustomerId(array $data): int
```

Comment:
making transformer just for one case is overhead, but yeah if you spotted at least 2 already, maybe we could homogenize it. I guess entity search input should automatically transform itself based on limit :thinking: 
But im already so out of bounds with this PR that I try to reject every unrelated change request, so please lets do it in another one, or this will never end :smile: